sdk/ml/azure-ai-ml/azure/ai/ml/operations/_evaluator_operations.py

```diff
@@ -0,0 +1,336 @@
+            Details will be provided in the error message.
+        :raises ~azure.ai.ml.exceptions.EmptyDirectoryError: Raised if local path provided points to an empty directory.
+        :return: Model asset object.
+        :rtype: ~azure.ai.ml.entities.Model
+        """
+        model.properties.update(_get_evaluator_properties())
+        return self._model_op.create_or_update(model)
+
+    def _raise_if_not_evaluator(self, properties: Optional[Dict[str, Any]], message: str) -> None:
+        """
+        :param properties: The properties of a model.
+        :type properties: dict[str, str]
+        :param message: The message to be set on exception.
+        :type message: str
+        :raises ~azure.ai.ml.exceptions.ValidationException: Raised if model is not an
+                                                             evaluator.
+        """
+        if properties is not None and not _is_evaluator(properties):
+            raise ResourceNotFoundError(
+                message=message,
+                response=None,
+            )
+
+    @monitor_with_activity(ops_logger, "Evaluator.Get", ActivityType.PUBLICAPI)
+    def get(self, name: str, version: Optional[str] = None, label: Optional[str] = None) -> Model:
+        """Returns information about the specified model asset.
+
+        :param name: Name of the model.
+        :type name: str
+        :param version: Version of the model.
+        :type version: str
+        :param label: Label of the model. (mutually exclusive with version)
+        :type label: str
+        :raises ~azure.ai.ml.exceptions.ValidationException: Raised if Model cannot be successfully validated.
+            Details will be provided in the error message.
+        :return: Model asset object.
+        :rtype: ~azure.ai.ml.entities.Model
+        """
+        model = self._model_op.get(name, version, label)
+
+        properties = None if model is None else model.properties
+        self._raise_if_not_evaluator(
+            properties,
+            f"Evaluator {name} with version {version} not found.",
+        )
+
+        return model
+
+    @monitor_with_activity(ops_logger, "Evaluator.Download", ActivityType.PUBLICAPI)
+    def download(self, name: str, version: str, download_path: Union[PathLike, str] = ".") -> None:
+        """Download files related to a model.
+
+        :param name: Name of the model.
+        :type name: str
+        :param version: Version of the model.
+        :type version: str
+        :param download_path: Local path as download destination, defaults to current working directory of the current
+            user. Contents will be overwritten.
+        :type download_path: Union[PathLike, str]
+        :raises ResourceNotFoundError: if can't find a model matching provided name.
+        """
+        self._model_op.download(name, version, download_path)
+
+    @monitor_with_activity(ops_logger, "Evaluator.Archive", ActivityType.PUBLICAPI)
+    def archive(
```

Comment:
My suggestion is to remove `archive`, `restore` and `share` functions in the first version. When evaluators are being added to GenericAssetService, not all functionality will be ready at once. So I'd prefer to only expose the most basic operations for now.

Reply:
Removed.